src/main/java/org/mastodon/mamut/treesimilarity/ZhangUnorderedTreeEditDistance.java

```diff
@@ -207,7 +234,34 @@ public static < T > double averageDistance( @Nullable final Tree< T > tree1, fin
 			final BiFunction< T, T, Double > costFunction )
 	{
 		double denominator = ( double ) TreeUtils.size( tree1 ) + ( double ) TreeUtils.size( tree2 );
-		// NB: avoid division by zero. Two empty trees are considered equal. Two trees with zero distance are considered equal.
+		// NB: avoid division by zero. Two empty trees are considered equal.
+		if ( denominator == 0 )
+			return 0;
+		return distance( tree1, tree2, costFunction ) / denominator;
+	}
+
+	/**
+	 * Calculates the average Zhang edit distance between two labeled unordered trees using averaging by the combined sum of the attribute values.
+	 * <br>
+	 * @see <a href="https://www.science.org/doi/suppl/10.1126/science.aar5663/suppl_file/aar5663_guignard_sm.pdf">Guignard et al. (2020) Page 38-39</a>
+	 *
+	 * @param tree1 Tree object representing the first tree.
+	 * @param tree2 Tree object representing the second tree.
+	 * @param costFunction mandatory cost function.
+	 * @param attributeSummarizer function to summarize the attribute values of the trees.
+	 * @param <T> Attribute type of the tree nodes.
+	 *
+	 * @return The average Zhang edit distance between tree1 and tree2.
+	 */
+	public static < T > double guignardAverageDistance(
+			@Nullable final Tree< T > tree1, final @Nullable Tree< T > tree2,
+			final BiFunction< T, T, Double > costFunction, final ToDoubleFunction< List< T > > attributeSummarizer
+	)
```

Comment:
Please remove the `costFunction` and `attributeSummarizer ` parameters, and simply allways use the default value `GUIGNARD_COST_FUNCTION` and `ATTRIBUTE_SUMMARIZER` there is no need to have this method that generalizable.

Reply:
I have removed now the whole separate guignard distance method

* It is not really needed, since normalization using attribute sums has been discarded after a discussion with the authors in favor of normalization using node sums
* Normalization using node sums has the same result as the existing normalized distance method with the guignard local cost function